### PR TITLE
Test against Django 2.2 with Travis build stages

### PR DESCRIPTION
**Pull request by bittner**
- Add Django 2.2 to the test matrix
- Remove all officially unsupported Python+Django combinations
- Add Bandit check, add Codacy badge
- Add docs target in Tox configuration
- Use build stages with Travis CI
- Automatically deploy when a tag is pushed

### CODE REVIEW

**Comment by bittner** [comment]
@jcassee Apart from Bandit, the security linter from PyCQA, I've added automatic deployment with this PR, which allows us to release a version by simply push in Git tag.

The latter currently uses my user name. We can change this to yours, you'd have to provide your encrypted password in this case (via the Travis CLI).

File: README.rst [hunk 1]
```diff
@@ -32,8 +32,8 @@ an asynchronous version of the Javascript code if possible.
 .. |coveralls| image:: https://coveralls.io/repos/jazzband/django-analytical/badge.svg
    :alt: Test coverage
    :target: https://coveralls.io/r/jazzband/django-analytical
-.. |health| image:: https://landscape.io/github/jazzband/django-analytical/master/landscape.svg?style=flat
-   :target: https://landscape.io/github/jazzband/django-analytical/master
+.. |health| image:: https://img.shields.io/codacy/grade/********************/master.svg
+   :target: https://www.codacy.com/app/jazzband/django-analytical
```
Comment (bittner) [inline README.rst]: Oooops! Can someone sign up django-analytical on Codacy? @jezdez maybe?

We can then get a proper badge from [the Codacy account](https://www.codacy.com/app/jazzband/django-analytical).

**Comment by jcassee** [comment]
> @jcassee Apart from Bandit, the security linter from PyCQA, I've added automatic deployment with this PR, which allows us to release a version by simply pushing a Git tag.
> 
> The latter currently uses my user name. We can change this to yours, you'd have to provide your encrypted password in this case (via the Travis CLI).

Cool, good idea! Using your username is fine, no need to change.

File: README.rst (hunk 1 above)
Comment (bittner) [reply README.rst]: This relates to #145, by the way.

**Pull request merged**